src/game_manager.py

```diff
@@ -180,11 +180,11 @@ def _clear_cell_value(self, cell: SudokuCell):
         UIHelpers.clear_feedback_classes(cell.get_style_context())
         self.game_board.set_input(row, col, None)
 
-    def _fill_cell(self, cell: SudokuCell, number: str):
+    def _fill_cell(self, cell: SudokuCell, number: str, state):
         UIHelpers.clear_conflicts(self.conflict_cells)
         row, col = cell.row, cell.col
 
-        if self.pencil_mode:
+        if self.pencil_mode or state:
```

Comment:
```suggestion
        if self.pencil_mode or ctrl_is_pressed:
```
I changed `state` to `ctrl_is_pressed` because I think it has better clarity.